src/api/app/models/project/signing_key_gpg.rb

```diff
@@ -15,7 +15,7 @@ class SigningKeyGPG
 
     def initialize(project_name)
       keyinfo = Xmlhash.parse(key_info_for_project(project_name))
-      @origin = keyinfo['project']
+      @origin = keyinfo['project'] || project_name
 
       return if keyinfo['pubkey'].blank?
```

Comment:
```suggestion
      return if keyinfo['pubkey'].blank?

      @origin = keyinfo['project'] || project_name
```

We could switch the order of the sentences. There is no need to do the assignment, if the condition is fullfiled.

Reply:
done